src/db.rs

```diff
@@ -626,6 +672,22 @@ impl Db {
         self.inner.delete_with_options(key, options).await;
     }
 
+    /// Write a batch of put/delete operations atomically to the database. All batch
+    /// operations are guaranteed to end up in the same WAL SST (or L0 memtable if
+    /// WAL is disabled).
+    pub async fn write_batch(&self, batch: &WriteBatch) {
+        self.inner
+            .write_batch_with_options(batch, DEFAULT_WRITE_OPTIONS)
+            .await;
+    }
+
+    /// Write a batch of put/delete operations atomically to the database. All batch
```

Comment:
Should the fact that `write_batch` would block writes and reads until the batch is written be added to the doc?

Reply:
Get takes a read lock ok the state. I wanted to know if it would be blocked, so I checked. I think would get blocked.

Reads being blocked does ensure correctness, at least for uncommitted isolation level, since otherwise it won't be atomic.

Reply:
Yep, it blocks gets right now since gets take a lock on the state.